src/utils.rs

```diff
@@ -0,0 +1,94 @@
+//! Utility functions for the Sprocket command line tool.
+
+use std::path::PathBuf;
+use std::fs;
+use std::io::Read;
+
+use anyhow::{Result, Context};
+
+/// Determines if a file is a YAML file based on its extension.
+pub fn is_yaml_file(path: &PathBuf) -> bool {
+    if let Some(extension) = path.extension() {
+        let ext = extension.to_string_lossy().to_lowercase();
+        return ext == "yaml" || ext == "yml";
+    }
+    false
+}
+
+/// Converts YAML content to JSON.
+pub fn convert_yaml_to_json(yaml_path: &PathBuf) -> Result<String> {
```

Comment:
Are you sure serde (or another crate) doesn't provide this functionality? YAML to JSON seems a common enough conversion I'd expect you to be able to pull a solution someone else wrote.